ext_mongodb.php

```diff
@@ -620,6 +620,78 @@ class BulkWriteException extends WriteException {}
 /* {{{ BSON and Serialization Classes */
 namespace MongoDB\BSON;
 
+/* {{{ Interfaces */
+interface TypeWrapper
+{
+	static public function createFromBSONType(\MongoDB\BSON\Type $type) : \MongoDB\BSON\TypeWrapper;
+	public function toBSONType() : \MongoDB\BSON\Type;
+}
+
+interface BinaryInterface
+{
+	public function __construct(string $data, int $type);
+	public function getType() : int;
+	public function getData() : string;
+	public function __debugInfo() : array;
+}
+
+interface Decimal128Interface
+{
+	public function __construct(string $decimal);
+	public function __toString() : string;
+	public function __debugInfo() : array;
+}
+
+interface JavascriptInterface
+{
+	public function __construct(string $code, ?mixed $scope = NULL);
+	public function __toString() : string;
+	public function __debugInfo() : array;
+	public function getCode() : string;
+	public function getScope() : mixed;
+}
+
+interface MaxKeyInterface
+{
+}
+
+interface MinKeyInterface
+{
+}
```

Comment:
Is there any value to defining either of these empty interfaces? Our classes don't even define `__toString()` methods for these.

There appears to be some debate over whether such interfaces are poor design. I suppose they might still be necessary since users can't extend the base classes and might need to use a type-hint on a method that requires an instance of one of these.